dataset/src/main/scala/frameless/functions/Udf.scala

```diff
@@ -0,0 +1,80 @@
+package frameless.functions
+
+import frameless.{TypedColumn, TypedEncoder}
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.functions.{udf => sudf}
+
+import scala.reflect.runtime.universe._
+
+trait Udf {
+  def udf[T, A: TypeTag, R: TypeTag](
+     f: A => R)
+  (implicit
+     aenc: TypedEncoder[A],
+     reng: TypedEncoder[R]):
+  (TypedColumn[T, A] => TypedColumn[T, R]) =
+    (u: TypedColumn[T, A]) => {
+      val uf = sudf(f)
```

Comment:
Vanilla Spark uses reflection to build encoders, that's why it needs `TypeTag`, frameless uses `TypedEncoder`, so we shouldn't need `TypeTag`. We can't reuse `org.apache.spark.sql.functions.udf` because it relies on reflection, but there is `UserDefinedFunction` expression that takes a list of input types that correspond to `TypedEncoder[A..Z].targetDataType`